### PR TITLE
Encryption methods

### PR DESCRIPTION
**Pull request by FPStruck**
Please run the code and let me know of any errors. I have tested as much as I can but I know there may be something I missed

### CODE REVIEW

File: UHDD/src/application/fxmlScenes/PatientInfoView.fxml
Comment (TitanSri) [inline UHDD/src/application/fxmlScenes/PatientInfoView.fxml:1]: This is an old file that is no longer in use

File: UHDD/src/application/viewControllers/DashboardController.java
Comment (TitanSri) [inline UHDD/src/application/viewControllers/DashboardController.java:1]: Verified that this scene is no longer seen in the application.

File: UHDD/src/application/viewControllers/PatientDirectoryController.java
Comment (TitanSri) [inline UHDD/src/application/viewControllers/PatientDirectoryController.java:1]: The structure of the main is different for the file location, ie. 
UHDD/src/application/viewControllers/application/viewControllers)
/PatientDirectoryController.java

File: UHDD/src/application/viewControllers/PatientInfoViewController.java
Comment (TitanSri) [inline UHDD/src/application/viewControllers/PatientInfoViewController.java:1]: Verified that this scene is no longer seen in the application.

**Review by TitanSri — DISMISSED** [review]

File: UHDD/src/application/fxmlScenes/Dashboard.fxml
Comment (TitanSri) [inline UHDD/src/application/fxmlScenes/Dashboard.fxml:1]: Works fine

File: UHDD/src/application/fxmlScenes/PatientDirectory.fxml
Comment (TitanSri) [inline UHDD/src/application/fxmlScenes/PatientDirectory.fxml:1]: Works fine

File: UHDD/src/application/DBConnector.java
Comment (TitanSri) [inline UHDD/src/application/DBConnector.java:1]: No longer in use on main

File: UHDD/src/application/viewControllers/DashboardController.java [hunk 1]
```diff
@@ -149,14 +149,17 @@ public void initialize() throws Exception{
...
-							if(patientDetails.next()) {
-								String name =  patientDetails.getString("FirstName") + " " 
-										+ patientDetails.getString("MiddleName") + " " 
-										+ patientDetails.getString("LastName");
-								fullName.setText(name);
-								phoneNumber.setText(patientDetails.getString("Telephone"));
-								pastMedicalConditions.setText(patientDetails.getString("PastMedicalConditions"));
-								progressNotes.setText(patientDetails.getString("ProgressNotes"));
+							if (patientDetails.next()) {
+							    String name =  patientDetails.getString("FirstName") + " "
```
Comment (TitanSri) [inline UHDD/src/application/viewControllers/DashboardController.java]: No longer in use in main

File: UHDD/src/application/viewControllers/PatientInformationController.java
Comment (TitanSri) [inline UHDD/src/application/viewControllers/PatientInformationController.java:1]: No longer in user in the application

File: UHDD/src/application/viewControllers/PatientInfoViewController.java [hunk 2]
```diff
@@ -78,25 +78,46 @@ public class PatientInfoViewController {
...
-			insuranceNumberTXT.setText(patientDetails.getString("HealthInsuranceNumber"));
-			detailsTXT.setText(patientDetails.getString("Details"));
-			cityTXT.setText(patientDetails.getString("City"));
-			emergencyNoTXT.setText(patientDetails.getString("EmergencyContactNumber"));
-			familyMedicalHistoryTXT.setText(patientDetails.getString("FamilyMedicalHistory"));
-			progressNotesTXT.setText(patientDetails.getString("ProgressNotes"));
-			
+        dbConnector.initialiseDB();
+        ResultSet patientDetails = dbConnector.QueryReturnResultsFromPatientId(Id);
+
```
Comment (TitanSri) [inline UHDD/src/application/viewControllers/PatientInfoViewController.java]: This is no longer in Mina's branch

File: UHDD/src/application/DBConnector.java [hunk 3]
```diff
@@ -207,20 +210,102 @@ public void executeQuery(String sql) throws SQLException {
...
-					statement.setString(5, role);
-					statement.setString(6, dateString);
-					statement.setString(7, dateString);
-					statement.executeUpdate();
-					
-				}
+		public void createUserExecuteQuery(String username, String passwordHash, String params, String email, String role, String dateString, String OTPSecretKey)
+		  throws SQLException {
+			String sql = "INSERT INTO user_details (username, password_hash, password_params, email, role, user_creation_date, password_last_modified, OTPSecretKey) VALUES (?, ?, ?, ?, ?, ?, ?, ?)";
+			PreparedStatement statement = connection.prepareStatement(sql);
```
Comment (TitanSri) [inline UHDD/src/application/DBConnector.java]: I failed to create a new user.
Error: "email failed to send"
Some one else needs to test the new user creation

File: UHDD/src/application/DBConnector.java [hunk 4]
```diff
@@ -207,20 +210,102 @@ public void executeQuery(String sql) throws SQLException {
...
+			statement.setString(6, dateString);
+			statement.setString(7, dateString);
+			statement.setString(8, OTPSecretKey);
+			statement.executeUpdate();
+			
+		}
+
+		public void createNewNoteExecuteQuery(String patientId, String doctorId, String noteText, String noteEnteredDate, String scriptIncluded)
+		  throws SQLException {
+			String sql = "INSERT INTO testdb.patient_notes (patientId, doctorId, noteText, noteEnteredDate, scriptIncluded) VALUES (?, ?, ?, ?, ?)";
```
Comment (TitanSri) [inline UHDD/src/application/DBConnector.java]: This is not working, this maybe the user session

File: UHDD/src/application/DBConnector.java [hunk 5]
```diff
@@ -207,20 +210,102 @@ public void executeQuery(String sql) throws SQLException {
...
+			statement.setString(2, doctorId);
+			statement.setString(3, noteText);
+			statement.setString(4, noteEnteredDate);
+			statement.setString(5, scriptIncluded);
+			statement.executeUpdate();
+			
+		}
+
+		public void createNewPatientExecuteQuery(
+			    String firstName, String middleName, String lastName, String gender,
```
Comment (TitanSri) [inline UHDD/src/application/DBConnector.java:243]: Create patient test fine

File: UHDD/src/application/DBConnector.java [hunk 6]
```diff
@@ -231,8 +316,50 @@ public void changePasswordExecuteQuery(String username, String passwordHash, Str
 					statement.setString(3, dateString);
 					statement.setString(4, username);
 					statement.executeUpdate();
-					
-				}
+			
+		}
+
+		public void CreateNewProgressPlanExecuteQuery(String diagnosisId, String initialDetails, String progressPlanGoal, String expectedRemediationDate) throws SQLException{
```
Comment (TitanSri) [inline UHDD/src/application/DBConnector.java]: Works fine

File: UHDD/src/application/DBConnector.java [hunk 7]
```diff
@@ -241,6 +368,28 @@ public ResultSet QueryReturnResultsFromUser(String username) throws SQLException
 
 	        return statement.executeQuery();
 	    }
+
+		public ResultSet QueryReturnResultsForUserSession(String username) throws SQLException {
```
Comment (TitanSri) [inline UHDD/src/application/DBConnector.java]: Works fine

File: UHDD/src/application/DBConnector.java [hunk 8]
```diff
@@ -249,32 +398,115 @@ public ResultSet QueryReturnResultsFromPatientId(String patientId) throws SQLExc
 			System.out.println(sql);
 	        return statement.executeQuery();
 	    }
+
+		public ResultSet QueryReturnResultsMedicationFromPatientId(String patientId) throws SQLException {
```
Comment (TitanSri) [inline UHDD/src/application/DBConnector.java]: Looks fine

File: UHDD/src/application/DataEncryptorDecryptor.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+public class DataEncryptorDecryptor {
```
Comment (TitanSri) [inline UHDD/src/application/DataEncryptorDecryptor.java:11]: tested fine

File: UHDD/src/application/viewControllers/application/viewControllers/PatientDirectoryController.java [hunk 10]
```diff
@@ -339,112 +337,11 @@ public void searchTable(MouseEvent mouseEvent) throws Exception {
 
 	        patientDirectorytableView.setItems(patientOL);
 	    }
-=======
```
Comment (TitanSri) [inline UHDD/src/application/viewControllers/application/viewControllers/PatientDirectoryController.java]: application seems to be navigating fine

**Review by TitanSri — COMMENTED** [review]
The new user creation is presenting the same issue with emailing the new user. However, it works fine on from the main. 
Apart from the new user creation, everything seems fine

**Review by TitanSri — CHANGES_REQUESTED** [review]
We need to find out why the email is not sending than I will approve the pull request

**Review by Losh19 — COMMENTED** [review]
Looks all good but idk if we should merge it until the decryption issue is sorted

**Comment by TitanSri** [comment]
> We need to find out why the email is not sending than I will approve the pull request

The Email Manager needs more parameters. I will push from Toby's branch

**Review by TitanSri — APPROVED** [review]
Looks good I got the email to work. 
The patient data are working

**Comment by TitanSri** [comment]
I am not sure why the paths skipped through empty folders for the view controllers.
UHDD/src/application/viewControllers/application/viewControllers

**Pull request merged**